File: scripts/php/j20.arguments.php
#!/usr/bin/env php
<?php
// illustrative variables
$arg_count = $argc - 1;  // get num of arguments
$first     = 1;          // set index of initial element
$last      = $arg_count; // set index of last element

echo "The arguments passed are (reverse order):\n";
// iterative loop to enumerate args
for ( $count = $last; $count >= $first; $count-- )
  // output count and argument suing count index
  echo "  item $count: $argv[$count]\n";
?>

File: gen_scripts/perl/j20.arguments.pl
#!/usr/bin/env perl -w
# illustrative variables
my $last  = $#ARGV; # get index of last element
my $first = 0;      # set index of first element

print "The arguments passed are (reverse order):\n";
# iterative loop to enumerate args
#   Notes:
#    - $count     is index into $ARGV list
#    - $count + 1 is count that is displayed to user
for ( my $count = $last; $count >= $first ; $count-- ) {
  my $arg = $ARGV[$count];  # get arg using index
  # output count and argument using count index
  printf "  item %d: %s\n", $count+1, $arg;
}
